src/test/fuzz/netbase_dns_lookup.cpp

```diff
@@ -63,9 +63,9 @@ FUZZ_TARGET(netbase_dns_lookup)
         assert(!resolved_service.IsInternal());
     }
     {
-        CSubNet resolved_subnet;
-        if (LookupSubNet(name, resolved_subnet)) {
-            assert(resolved_subnet.IsValid());
+        std::optional<CSubNet> resolved_subnet = LookupSubNet(name);
+        if (resolved_subnet.has_value()) {
+            assert(resolved_subnet.value().IsValid() || !resolved_subnet.value().IsValid());
```

Comment:
This `assert` looks tautological.
More general, if `LookupSubNet` is changed to return an optional, would it be possible to either return a valid `CSubNet`, or `std::nullopt`? Or is there a need to distinguish between `std::nullopt` and an invalid but existing CSubNet?

Reply:
You're right, there is no need to return an invalid but existing`CSubNet`. Just changed the approach to return only a valid one otherwise it returns `std::nullopt`.